test/unit/eslint-plugin-next/no-html-link-for-pages.test.ts

```diff
@@ -186,15 +212,47 @@ export class Blah extends Head {
 `
 
 describe('no-html-link-for-pages', function () {
-  it('valid link element', function () {
-    const report = linter.verify(validCode, linterConfig, {
+  it('prints warning when there are no "pages" or "app" directories', function () {
+    const consoleSpy = jest.spyOn(console, 'warn').mockImplementation()
+    withoutPagesLinter.verify(validCode, linterConfig, {
       filename: 'foo.js',
     })
+    const rootDirectory = path.join(__dirname, 'without-pages-dir')
+    expect(consoleSpy).toHaveBeenCalledWith(
+      `Pages directory cannot be found at ${path.join(
+        rootDirectory,
+        'pages'
+      )} or ${path.join(
+        rootDirectory,
+        'src',
+        'pages'
+      )}. If using a custom path, please configure with the \`no-html-link-for-pages\` rule in your eslint config file.`
+    )
+
+    consoleSpy.mockRestore()
+  })
+  it('does not print warning when there is "app" directory and no "pages" directory', function () {
```

Comment:
I noticed there's something wrong going on with this test suite. When trying to make it fail and running the only this suite, it fails (as it should). But when running whole test file, this test passes and results in false positive. Probably has something to do with mocking, but I couldn't figure out the exact problem.